src/init/features/apphosting/repo.ts

```diff
@@ -92,11 +92,11 @@ export async function linkGitHubRepository(
 ): Promise<gcb.Repository> {
   utils.logBullet(clc.bold(`${clc.yellow("===")} Set up a GitHub connection`));
   // Fetch the sentinel Oauth connection first which is needed to create further GitHub connections.
+  await ensureSecretManagerAdminGrant(projectId);
```

Comment:
This change requires all callers of this command to have access to fetch the IAM policy (to ensure that the CB service account has SecretManager permissions), which I don't believe is intentional.

We probably want to refactor getOrCreateOauthConnection to ensure the grant prior to creating the oauth connection if it doesnt exist

Reply:
ah good point, ya we only want to check for this when users go to create an oauth connection.